compiler-cli/src/dependencies.rs

```diff
@@ -285,9 +286,31 @@ pub fn download<Telem: Telemetry>(
     }
     LocalPackages::from_manifest(&manifest).write_to_disc(paths)?;
 
+    let package_fetcher = PackageFetcher::boxed(runtime.handle().clone());
+    let major_versions_available =
+        dependency::check_for_major_version_updates(&manifest, package_fetcher);
+    if !major_versions_available.is_empty() {
+        pretty_print_major_versions_available(major_versions_available);
+    }
+
     Ok(manifest)
 }
 
+fn pretty_print_major_versions_available(versions: dependency::PackageVersionDiffs) {
+    let longest_package_name_length = versions.keys().map(|name| name.len()).max().unwrap_or(10);
+
+    // print to stderr instead of stdout because this is not part of the standard output of this
+    // command
+    eprintln!("\nHint: the following dependencies have new major versions available...\n");
+    for (name, (v1, v2)) in versions {
+        let padding = " ".repeat(longest_package_name_length - name.len());
+
+        // lazily assuming the longest version could be 8 characters. eg: 1.1234.2
+        // excluding qualifiers other than x.y.z
+        eprintln!("{name}:{padding} {v1:<8} -> {v2:<8}");
```

Comment:
Let's render this to a string and then snapshot test the function. This function can then just print the string and not be responsible for formatting and printing

Reply:
Done.